axelor-base/src/main/java/com/axelor/apps/base/service/PartnerService.java

```diff
@@ -514,4 +516,39 @@ public String computeCompanyStr(Partner partner) {
     }
     return null;
   }
+
+  public Partner isThereDuplicatePartnerInArchive(Partner partner) {
+    String newName = this.computeSimpleFullName(partner);
+    if (Strings.isNullOrEmpty(newName)) {
+      return null;
+    }
+    Long partnerId = partner.getId();
+    if (partnerId == null) {
+      Partner existingPartner =
+          partnerRepo
+              .all()
+              .filter(
+                  "lower(self.simpleFullName) = lower(:newName) "
+                      + "and self.partnerTypeSelect = :_partnerTypeSelect "
+                      + "and self.archived = true")
+              .bind("newName", newName)
+              .bind("_partnerTypeSelect", partner.getPartnerTypeSelect())
+              .fetchOne();
+      return existingPartner;
+    } else {
+      Partner existingPartner =
+          partnerRepo
+              .all()
+              .filter(
+                  "lower(self.simpleFullName) = lower(:newName) "
+                      + "and self.id != :partnerId "
+                      + "and self.partnerTypeSelect = :_partnerTypeSelect "
+                      + "and self.archived = true")
+              .bind("newName", newName)
+              .bind("partnerId", partnerId)
+              .bind("_partnerTypeSelect", partner.getPartnerTypeSelect())
+              .fetchOne();
+      return existingPartner;
+    }
+  }
```

Comment:
Having almost the same query written 4 times will be very difficult to maintain. Maybe it is possible to check whether existingPartner returned by isThereDuplicatePartner is archived.